evm/rlp/headers.py

```diff
@@ -247,7 +247,7 @@ def hash(self) -> bytes:
                 int_to_bytes32(self.number),
             ))
         )
-        return header_hash
+        return pad32(header_hash[8:])
```

Comment:
Will we keep using the padded `bytes32` from `bytes24` in the future? or is it just temporary

Reply:
Dealing with `bytes24` as header hash(e.g., `collation_headers: public(bytes32[bytes <= 24][int128])`) in Vyper takes extra ~30000 gas in `addHeader`.
Perhaps pad it to 32 bytes in contract while remain 24 bytes outside out contract? Though it could be confusing for other developers since this is a Vyper specific issue.

Reply:
yes it makes sense.